text/0000-build-std.md

```diff
@@ -0,0 +1,2390 @@
+  parameters**, [jschwe], Jun 2020
+  - User wants to have one package depend on `core` and `alloc` and another
+    package depend on `core`, `alloc` and `std` but this isn't possible with the
+    current experimental flag
+  - Closed as duplicate of [wg-cargo-std-aware#5]
+- **[wg-cargo-std-aware#68]: Support Profile Guided Optimisation (PGO)**, [errantmind], Mar 2021
+  - `profiler_builtins` isn't compatible with `no_core`
+    - i.e. `profiler_builtins` requires `core` so if building with profiling, need to
+      build `core` first w/out the profiling instrumentation
+  - [rust#79958] improves the error message for this
+- **internals.r-l.o: [Build-std and the standard library][wg-cargo-std-aware#43-internals]**, [ehuss], Dec 2019
+  - Proposes changing the standard library so all of its crates can build on all
+    targets
+    - It may not work and not all APIs will be available, but there will be no
+      compilation errors
+      - i.e. add runtime errors or cfgs
+    - Makes things simpler because build-std can become a simple toggle
+  - Building an empty standard library is preferred over not building at all as
+    this avoids breaking `no_std` crates that build for tier three targets with
+    no standard library support at all
+  - Suggestion that the standard library use Cargo features more so that most
+    functionality is gated by default
+    - Fixes this issue as most of the standard library would be absent by
+      default
+  - Some prefer that the standard library be listed in the `Cargo.toml`
+    eventually but that this would work initially
+    - Explicit standard library dependencies are being considered
+      - It could be a inconvenience to need to specify the standard library
+        dependencies in most cases
+        - Most packages don't need it
+      - Need default for packages that do not specify a standard library
+        dependency
+  - References to the "portability lint"
+    - Lint that triggers when calling an available platform-specific code so
+      that users can avoid making code less portable
+    - Lint was later deemed infeasible
+  - Described as necessary step towards long-standing goal to move from a
+    multiple-crates model of the standard library to a cfg-flag model
+  - Cargo could learn what targets a package supports
+    - e.g. libstd could declare this
+    - Doesn't work for custom targets (i.e. target-spec-json)
+    - Could accidentally allow the standard library to be used in `no_std`
+      projects
+    - Adds noise to `Cargo.toml`
+- **[wg-cargo-std-aware#85]: Figure out rust-lang/rust testing strategy**, [ehuss], Mar 2023
+  - How to test build-std when it is tightly coupled to the standard library source tree?
+    - Could use rust-toolstate but wasn't very good
+    - Could use git subtrees (or [JOSH])
+    - Could aim to reduce coupling between the standard library and Cargo
+  - Which targets get tested and how?
```

Comment:
"Which targets" has kind of been decided by the Target Tier Policy right?

Reply:
Yes, I think so. I was just summarizing what had been written in the issues with this section, rather than adding commentary/up-to-date information. I've added a note to mention that - the appendix is just a summary of issues, and then when referenced in the main body of the RFC, will be accompanied with up-to-date information.